taniumrest_connector.py

```diff
@@ -940,94 +988,119 @@ def _handle_get_question_results(self, param):
+            error_code, error_msg = self._get_error_message_from_exception(e)
+            self.debug_print(
+                "Error occurred while converting the sensors. Error code: {}, Message: {}".format(error_code, error_msg))
+            return
 
-            raw_parameter_definition = resp_data.get("parameter_definition", "")
-            parameter_definition = None
-            try:
-                if raw_parameter_definition:
-                    parameter_definition = json.loads(raw_parameter_definition)
-            except Exception as e:
-                error_code, error_msg = self._get_error_message_from_exception(e)
-                return action_result.set_status(phantom.APP_ERROR,
-                                                "Error while parsing the 'parameter_definition'. Error Code: {0}. Error Message: {1}".format(error_code, error_msg))
-
-            if parameter_definition:
-                # Parameterized Sensor
-                parameter_keys = [parameter["key"] for parameter in parameter_definition["parameters"]]
-                self.save_progress("Parameter Keys:\n{}".format(json.dumps(parameter_keys)))
-                total_params += len(parameter_keys)
-                parameters = []
-
-                for key in parameter_keys:
-                    if param_idx >= len(param_list):
-                        action_result.set_status(phantom.APP_ERROR, "For parameters which you do not want to add value, please use double quotes(\"\").\
-                                    For more details refer to the documentation")
-                        return
-
-                    parameter = {
-                        "key": "||%s||" % key,
-                        "value": param_list[param_idx] }
-                    parameters.append(parameter)
-                    param_idx += 1
-
-                sensor_dict = {
-                    "source_hash": sensor["hash"],
-                    "parameters": parameters }
-                sensor_data.append({"sensor": sensor_dict})
-            else:
-                # Regular Sensor, can use as-is
-                sensor_data.append({"sensor": sensor})
+        self.save_progress("Sensor Data:\n{}".format(json.dumps(sensor_data)))
 
-        if total_params and total_params != len(param_list):
+        if self._param_idx and self._param_idx != len(param_list):
             action_result.set_status(phantom.APP_ERROR, "Please provide the exact number of parameters expected by the sensor")
             return
 
-        self.save_progress("Sensor Data:\n{}".format(json.dumps(sensor_data)))
-        question_data = { "selects": sensor_data,
-                          "question_text": query["question_text"]}
-        if "group" in query:
-            # Add in filters
-            question_data["group"] = query["group"]
-
         return question_data
 
+    def _create_sensor_dict(self, action_result, sensor, param_list):
+        sensor_name = sensor["name"]
```

Comment:
Would it be possible to add some comments explaining the major steps? (e.g. `create sensor dict from ...`)

Reply:
Added a doc-string to the '_load_full_sensors_to_obj' and '_create_sensor_dict' functions.